### PR TITLE
rquickshare: init at 0.7.1

### LINKED ISSUE #303967: Package request: rquickshare
**Project description**

[RQuickShare](https://github.com/Martichou/rquickshare) is implementation of QuickShare (NearbyShare)

About QuickShare:
- https://support.google.com/android/answer/9286773
- https://blog.google/products/android/ces-2024-android-updates

**Metadata**

* homepage URL: https://github.com/Martichou/rquickshare
* source URL: https://github.com/Martichou/rquickshare
* license: gpl3
* platforms: linux

> rquickshare is still in development (WIP) and currently only supports Linux even tho it should be compatible with macOS too. 

It might also work on macOS.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by luftmensch-luftmensch**
## Description of changes
Closes #303967

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by luftmensch-luftmensch** [comment]
```
> nix build .#rquickshare
> tree result/
 result
└──  bin
   └──  rquickshare
```

File: pkgs/by-name/rq/rquickshare/package.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      hash = "sha256-716d7T4nbs/dDS4KVGTADCpLO31U8iq6hDVD+c7Ks1I=";
+    };
+
+    meta = with lib; {
+      description = "Rust implementation of NearbyShare/QuickShare from Android for Linux";
+      homepage = "https://github.com/Martichou/rquickshare";
+      changelog = "https://github.com/Martichou/rquickshare/blob/v${version}/CHANGELOG.md";
+      license = licenses.agpl3Plus;
+      maintainers = with maintainers; [luftmensch-luftmensch];
+      platforms = ["x86_64-linux"];
```
Comment (Qyriad) [inline pkgs/by-name/rq/rquickshare/package.nix:22]:
```suggestion
      platforms = ["x86_64-linux"];
      sourceProvenance = with sourceTypes; [ binaryNativeCode ];
```

**Review by Qyriad — CHANGES_REQUESTED** [review]
This file should also probably be formatted with `pkgs.nixfmt-rfc-style`

File: pkgs/by-name/rq/rquickshare/package.nix (hunk 1 above)
Comment (luftmensch-luftmensch) [reply pkgs/by-name/rq/rquickshare/package.nix]: Applied suggestion!

**Review by Qyriad — COMMENTED** [review]
LGTM, thanks!

File: pkgs/by-name/rq/rquickshare/package.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  src = fetchurl {
+    url = "https://github.com/Martichou/rquickshare/releases/download/v${version}/r-quick-share_${version}_amd64.AppImage";
+    hash = "sha256-716d7T4nbs/dDS4KVGTADCpLO31U8iq6hDVD+c7Ks1I=";
+  };
+
+  meta = {
+    description = "Rust implementation of NearbyShare/QuickShare from Android for Linux";
+    homepage = "https://github.com/Martichou/rquickshare";
+    changelog = "https://github.com/Martichou/rquickshare/blob/v${version}/CHANGELOG.md";
+    license = lib.licenses.agpl3Plus;
```
Comment (honnip) [inline pkgs/by-name/rq/rquickshare/package.nix:21]: https://github.com/Martichou/rquickshare/blob/master/LICENSE  

It's `gpl3Plus`

**Review by honnip — COMMENTED** [review]
We can also use desktop entry and icon from the source.

File: pkgs/by-name/rq/rquickshare/package.nix (hunk 2 above)
Comment (luftmensch-luftmensch) [reply pkgs/by-name/rq/rquickshare/package.nix]: Applied suggestion/fixes!

Comment (honnip) [reply pkgs/by-name/rq/rquickshare/package.nix]: It needs to replace `Exec=r-quick-share` with `Exec=rquickshare` in the desktop entry

Comment (luftmensch-luftmensch) [reply pkgs/by-name/rq/rquickshare/package.nix]: Apologies for keeping you waiting. Patched the desktop entry.

Should I also rename the desktop entry file from `r-quick-share.desktop` to `rquickshare.desktop`?

Comment (honnip) [reply pkgs/by-name/rq/rquickshare/package.nix]: > `r-quick-share.desktop` to `rquickshare.desktop`

It's okay to leave it untouched

And it's now `Icon=rquickshare %u`, so the icon is broken. You might want to replace that line `Exec=r-...`, not `r-quick-share`

Comment (luftmensch-luftmensch) [reply pkgs/by-name/rq/rquickshare/package.nix]: Should be fixed now

**Comment by luftmensch-luftmensch** [comment]
@honnip Do you know what is causing the failure of  `ofborg-eval-lib-tests` ?

**Comment by honnip** [comment]
@ofborg eval

**Comment by honnip** [comment]
Result of `nixpkgs-review pr 311433` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>rquickshare</li>
  </ul>
</details>

**Review by honnip — APPROVED** [review]

**Pull request merged**